Event/registration PHP 8 compatibility improvements

JKingsnorth opened the pull request:
Before
----------------------------------------

On 'Manage Events':

`Deprecated function: strtolower(): Passing null to parameter #1 ($string) of type string is deprecated in CRM_Event_Page_ManageEvent->browse() (line 285 of...modules/contrib/civicrm/CRM/Event/Page/ManageEvent.php).`

On event registration forms:

`Deprecated function: trim(): Passing null to parameter #1 ($string) of type string is deprecated in CRM_Core_BAO_CustomField::setProfileDefaults() (line 1347 of ...modules/contrib/civicrm/CRM/Core/BAO/CustomField.php).`


After
----------------------------------------
Fewer PHP warnings

Technical Details
----------------------------------------

- Check if empty before the strtolower call
- Also removed an unused variable in the 'browse' method.

JKingsnorth committed:
Fix PHP 8 deprecated function notices

seamuslee001 commented on CRM/Event/Page/ManageEvent.php:
```diff
@@ -281,8 +281,8 @@ public function browse() {
       'String',
       $this
     );
-    $createdId = CRM_Utils_Request::retrieve('cid', 'Positive', $this, FALSE, 0);
```
Just wondering if this is something to do with search not sure why your removing this

JKingsnorth replied:
It seems to be un-used in the browse() function?

JKingsnorth commented:
Thanks for the eyes on this @seamuslee001 - I hoped to do a bit more but I've had to switch on to something else. Removed the bad commit, and squashed the others.
